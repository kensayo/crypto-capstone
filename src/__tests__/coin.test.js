import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configStore';
import Coin from '../components/coin';

describe('List all coins', () => {
  test('should container the list container', () => {
    render(
      <Provider store={store}>
        <Router>
          <Coin />
        </Router>
      </Provider>,
    );
    return expect(screen.getByTestId('list-container')).toBeInTheDocument;
  });
});

describe('Compare with snapshot', () => {
  it('render correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Coin />
          </Router>
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
