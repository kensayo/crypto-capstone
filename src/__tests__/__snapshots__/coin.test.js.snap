// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compare with snapshot render correctly 1`] = `
<main>
  <header
    className="d-flex flex-row align-items-center"
  >
    <img
      alt="Logo Img"
      className="w-50 p-3"
      src="/img/logo.png"
    />
    <div
      className="p-3"
    >
      <h1
        className="fw-bold rs-f text-uppercase"
      >
        Total 24h Vol
      </h1>
      <h3
        className="rs-f3"
      >
        $0.00
      </h3>
    </div>
  </header>
  <div
    className="articleHeader p-1 w-100"
  >
    <p
      className="m-0"
    >
      COIN DETAILS
    </p>
  </div>
  <article
    className="coinsContainer"
    data-testid="list-container"
  />
</main>
`;
